cmd/kubeadm/app/cmd/join.go

```diff
@@ -339,28 +384,6 @@ func newJoinData(cmd *cobra.Command, args []string, opt *joinOptions, out io.Wri
 		opt.externalcfg.ControlPlane = nil
 	}
 
-	// if the admin.conf file already exists, use it for skipping the discovery process.
-	// NB. this case can happen when we are joining a control-plane node only (and phases are invoked atomically)
-	var adminKubeConfigPath = kubeadmconstants.GetAdminKubeConfigPath()
```

Comment:
It would be better for testing if this is supplied as a func parameter. That way we can have more comprehensive tests, and we won't need to override `constants.KubernetesDir` to test this func.

Reply:
good idea, I revised.